### PR TITLE
cmake: enable x86_64 -> i386 cross-compilation

### PR DESCRIPTION
**Pull request by matthiasdiener**
Followup of #3332.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -85,6 +85,13 @@ set(NETWORK "netlrts" CACHE STRING "Target network layer, can be one of mpi,mult
 set(TARGET "charm++" CACHE STRING "Target build type, can be one of charm++,AMPI,LIBS,charm4py,ChaNGa,everylb,all-test")
 set(ARCH "" CACHE STRING "Target architecture, can be one of i386,x86_64,arm7,arm8,ppc64le") # By default, detect arch we are running on.
 
+if(NOT ${ARCH} STREQUAL ${CMAKE_HOST_SYSTEM_PROCESSOR})
```
Comment (evan-charmworks) [inline CMakeLists.txt:88]: Does the statement work without the `${}`?

File: CMakeLists.txt [hunk 2]
```diff
@@ -85,6 +85,13 @@ set(NETWORK "netlrts" CACHE STRING "Target network layer, can be one of mpi,mult
...
+  message(WARNING "Attempting to cross-compile from ${CMAKE_HOST_SYSTEM_PROCESSOR} to ${ARCH}. This is only supported for x86_64 -> i386 currently.")
+  set(CMAKE_C_FLAGS -m32)
+  set(CMAKE_CXX_FLAGS -m32)
+  set(CMAKE_FORTRAN_FLAGS -m32)
```
Comment (evan-charmworks) [inline CMakeLists.txt:92]: Could these be inside a test that ARCH == i386?

File: CMakeLists.txt (hunk 1 above)
Comment (matthiasdiener) [reply CMakeLists.txt]: Fixed.

File: CMakeLists.txt (hunk 2 above)
Comment (matthiasdiener) [reply CMakeLists.txt]: Fixed

**Comment by matthiasdiener** [comment]
Tested manually on `charity`, seems to work fine.

**Review by evan-charmworks — APPROVED** [review]

**Review by stwhite91 — APPROVED** [review]

**Pull request merged**